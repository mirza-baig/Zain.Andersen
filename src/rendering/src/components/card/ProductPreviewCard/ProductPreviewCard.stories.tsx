// Global
import { Story, Meta } from '@storybook/react';
// Local
import ProductPreviewCard, { ProductPreviewCardProps } from './ProductPreviewCard';
import defaultData from './ProductPreviewCard.mock-data';
import classNames from 'classnames';
import { Component } from 'src/helpers/Component';

export default {
  title: 'Card/ProductPreviewCard',
  component: ProductPreviewCard,
} as Meta;
const Template: Story<ProductPreviewCardProps> = (props) => {
  return (
    <Component variant="lg" dataComponent="listing/xupcardcollection" {...props}>
      <div className={classNames('col-span-4')}>
        <ProductPreviewCard {...props} />
      </div>
      <div className={classNames('col-span-4')}>
        <ProductPreviewCard {...props} />
      </div>
      <div className={classNames('col-span-4')}>
        <ProductPreviewCard {...props} />
      </div>
    </Component>
  );
};

export const Default = Template.bind({});
Default.args = defaultData;
