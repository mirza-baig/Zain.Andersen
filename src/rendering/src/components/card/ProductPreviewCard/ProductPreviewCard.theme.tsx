// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const ProductPreviewCardTheme: ThemeFile = {
  aw: {
    classes: {
      /**  Theme classes  **/
      productPreviewCradWrapper: 'flex flex-col items-start border border-gray h-full relative',
      eyebrow: 'text-dark-gray uppercase text-xxs font-heavy order-1',
      headlineWrapper: 'order-2 font-heavy font-sans text-xs',
      headline: 'text-xs font-heavy hover:underline ',
      ratingsAndPriceWrapper: 'flex items-center order-3',
      ratingsIconsList: 'flex mr-xxs',
      ratingsText: 'pr-xxs text-xs',
      priceLevelWrapper: 'border-l border-secondary px-xxs leading-none flex',
      priceTextClasses: 'font-sans text-gray text-xs',
      priceLevelClasses: 'text-black font-heavy',
      body: 'px-s my-s order-3 text-dark-gray',
      headerWrapper: 'flex flex-col items-start px-s pb-s pt-m gap-2 order-1 w-full',
      awColorsandCTA:
        'flex max-lg:flex-wrap  lg:mb-s lg:items-end justify-between w-full mt-auto max-lg:mb-s order-3',
      colorSwatchesWrapper: 'px-xs max-lg:mb-s',
      colorLabel: 'text-small',
      swatches: 'mt-xxxs flex',
      actions: 'px-xxs md:px-s order-5 max-md:[&_a]:px-xxs [&_a]:mr-0 max-md:mt-auto',
      buttonGroupClass: {
        wrapper: 'mb-0',
        cta1Classes: 'mb-0',
      },
      colorSwatches:
        'h-[30px] w-[30px] rounded-full mr-[14px] [&_*]:h-[30px] [&_*]:w-[30px] [&_img]:rounded-full',
      favoriteProduct:
        'favorite-product absolute right-0 top-0 inline h-0 w-0 cursor-pointer border-t-0 border-l-0 border-r-[60px] border-b-[60px] border-solid border-[transparent_#e3e3e3_transparent_transparent] transition-[border-color]  duration-500 ease-[ease]',
      favoriteIcon: 'absolute -right-[51px] top-[11px]',
    },
  },
  rba: {
    classes: {
      /**  Theme classes **/
      productPreviewCradWrapper: 'flex h-full flex-col items-start relative',
      eyebrow: 'text-xs text-dark-gray font-semi-bold order-2',
      headlineWrapper: 'order-1 text-s font-sans font-medium w-full',
      headline: 'text-s font-medium font-sans hover:underline max-w-[calc(100%_-_22px)]',
      ratingsAndPriceWrapper: 'flex order-3 items-center pt-xxs',
      ratingsIconsList: 'flex mr-xxs ',
      ratingsText: 'pr-xxs text-xs',
      priceLevelWrapper: 'border-l border-grey px-xxs leading-none flex',
      priceTextClasses: 'font-sans text-gray text-base font-medium',
      priceLevelClasses: 'text-black',
      body: 'mb-s order-4 text-dark-gray',
      headerWrapper: 'flex flex-col items-start pt-xs mb-s border-t border-black w-full order-1',
      colorSwatchesWrapper: 'my-s order-3 flex items-center w-full',
      colorLabel: 'text-body text-secondary font-bold',
      swatches: 'mt-xxxs ml-xxs flex',
      actions: 'border-b border-gray w-full md:pb-xs order-5 mt-auto',
      buttonGroupClass: {
        wrapper: 'flex-col md:items-center flex-wrap',
        cta1Classes: 'mr-2',
        cta2Classes: 'my-s lg:my-0',
      },
      colorSwatches:
        'h-[24px] w-[24px] rounded-full mr-xxs [&_*]:h-[24px] [&_*]:w-[24px] [&_img]:rounded-full',
      favoriteProduct:
        'favorite-product absolute right-0 top-[1px] inline h-0 w-0 cursor-pointer duration-500 ease-[ease]',
      favoriteIcon: 'absolute right-[0px] top-[12px]',
    },
  },
};
