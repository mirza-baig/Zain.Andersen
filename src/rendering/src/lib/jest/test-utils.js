// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
