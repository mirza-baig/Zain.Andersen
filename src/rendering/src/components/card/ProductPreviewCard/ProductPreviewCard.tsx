// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { ThemeName, useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck, Text } from '@sitecore-jss/sitecore-jss-nextjs';
import Link from 'next/link';
import classNames from 'classnames';

// Components
import { ProductPreviewCardTheme } from './ProductPreviewCard.theme';
import { Headline } from 'src/helpers/Headline';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { Eyebrow } from 'src/helpers/Eyebrow';
import { ButtonGroup } from 'src/helpers/ButtonGroup';
import { ImagePrimary, Image } from 'src/helpers/Media';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { EnumField } from 'lib/utils/get-enum';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';
import SingleButton from 'src/helpers/SingleButton/SingleButton';
import PriceLevel from 'src/helpers/PriceLevel/PriceLevel';
import { useBVScript } from 'lib/utils/use-bv-script';

type priceLevel = 1 | 2 | 3 | 4 | 5;

type ProductItem = Feature.EnterpriseWeb.Enterprise.Data.Products.EnterpriseProduct & {
  fields?: {
    standardExteriorColors?: Feature.EnterpriseWeb.Enterprise.Elements.Swatches.SwatchCollection & {
      fields?: {
        swatches: Feature.EnterpriseWeb.Enterprise.Elements.Swatches.Swatch[];
      };
    };
    priceLevel: EnumField<priceLevel>;
    featuredExteriorColors: Feature.EnterpriseWeb.Enterprise.Elements.Swatches.Swatch[];
  };
};

export type ProductPreviewCardProps = Feature.EnterpriseWeb.Enterprise.Cards.ProductPreviewCard & {
  fields?: {
    productItem: ProductItem;
  };
};

const ProductPreviewCard = (props: ProductPreviewCardProps) => {
  const { fields } = props;
  const { themeName, themeData } = useTheme(ProductPreviewCardTheme);
  const favoriteProductsArr = props.favoriteProducts || [];

  // Add the bazaarvoice script
  useBVScript();
  const bazaarvoiceProductId = fields?.productItem?.fields?.bazaarvoiceProductId?.value;
  if (!fields) {
    return null;
  }

  const showFavorite = fields?.favorite?.value;
  const productID = fields.productItem?.fields?.productId?.value;
  const isFavorited = favoriteProductsArr.includes(productID);

  const colorSwatchesAll = fields.productItem?.fields?.standardExteriorColors?.fields?.swatches;
  const colorSwatchesTotal = colorSwatchesAll && colorSwatchesAll.length;

  let featuredExteriorColors = fields?.productItem?.fields?.featuredExteriorColors || [];

  let colorSwatchesCount = 0;

  const MAX_FEATURE_EXTERIOR_COLORS: Record<ThemeName, number> = {
    aw: 3,
    rba: 5,
  };

  if (colorSwatchesTotal && featuredExteriorColors?.length > 0) {
    const maxColors = MAX_FEATURE_EXTERIOR_COLORS[themeName];
    featuredExteriorColors = featuredExteriorColors.slice(0, maxColors);
    colorSwatchesCount = colorSwatchesTotal - featuredExteriorColors.length;
  }

  const priceLevel =
    fields?.productItem?.fields?.priceLevel?.fields?.priceLevelText?.value &&
    (+fields?.productItem?.fields?.priceLevel?.fields?.priceLevelText?.value as priceLevel | null);

  const ColorAndActions = (): JSX.Element => {
    return (
      <>
        {featuredExteriorColors?.length > 0 && (
          <div className={themeData.classes.colorSwatchesWrapper}>
            <Text
              tag={'h4'}
              field={{ value: 'Exterior Colors' }}
              className={themeData.classes.colorLabel}
            />
            <Link
              href={
                fields.cta1Link.value.href ||
                fields.productItem?.fields?.productDetailPageLink?.value?.href ||
                ''
              }
            >
              <a>
                <div className={themeData.classes.swatches}>
                  {featuredExteriorColors?.map(
                    (
                      color: Feature.EnterpriseWeb.Enterprise.Elements.Swatches.Swatch,
                      index: number
                    ) => {
                      return (
                        <div key={index} className={themeData.classes.colorSwatches}>
                          <Image image={color.fields?.swatchImage} />
                        </div>
                      );
                    }
                  )}
                  {colorSwatchesCount > 0 && <div>+{colorSwatchesCount}</div>}
                </div>
              </a>
            </Link>
          </div>
        )}
        <div className={themeData.classes.actions}>
          {fields.cta1Link?.value?.href && fields.cta1Link?.value?.text ? (
            <ButtonGroup classes={themeData.classes.buttonGroupClass} {...props} />
          ) : fields.productItem?.fields.productDetailPageLink?.value?.text ? (
            <SingleButton
              classes={
                (themeData.classes.buttonGroupClass,
                { wrapper: 'mb-0 md:mb-s cursor-pointer', cta1Classes: 'font-bold' })
              }
              fields={{
                cta1Link: {
                  value: {
                    href: fields.productItem.fields.productDetailPageLink.value.href,
                    text: fields.productItem.fields.productDetailPageLink.value.text,
                    anchor: fields.productItem.fields.productDetailPageLink.value.anchor,
                    target: fields.productItem.fields.productDetailPageLink.value.target,
                  },
                },
                cta1AriaLabel: {
                  value: '',
                },
                cta1ModalLinkText: {
                  value: '',
                },
                cta1Style: {
                  id: '',
                  url: '',
                  name: 'Primary',
                  displayName: 'Primary',
                  fields: {
                    Value: {
                      value: fields.cta1Style?.fields?.Value?.value || 'primary',
                    },
                  },
                  templateId: '',
                  templateName: 'Enum',
                },
                cta1Icon: {
                  id: '',
                  url: '',
                  name: 'Arrow',
                  displayName: 'Arrow',
                  fields: {
                    Value: {
                      value: 'arrow',
                    },
                  },
                  templateId: '',
                  templateName: 'Enum',
                },
              }}
            />
          ) : (
            <></>
          )}
        </div>
      </>
    );
  };

  let primaryImageVal = {};
  let primaryImageMobileVal = {};

  const primaryImage = fields.primaryImage?.value?.src && fields.primaryImage.value;
  const primaryImageMobile =
    fields.primaryImageMobile?.value?.src && fields.primaryImageMobile?.value;
  const productImage = fields.productItem?.fields?.productImage?.value;
  const productImageMobile = fields.productItem?.fields?.productImageMobile?.value;

  primaryImageVal = primaryImage || productImage || {};
  primaryImageMobileVal = primaryImageMobile || productImageMobile || primaryImageVal;

  return (
    <div
      className={themeData.classes.productPreviewCradWrapper}
      data-component="card/productpreview"
    >
      {/* Favourite */}
      {showFavorite && productID && (
        <div className="absolute top-0 right-0">
          <div
            className={classNames(
              themeData.classes.favoriteProduct,
              isFavorited ? 'favorited border-[transparent_#f26924_transparent_transparent]' : ''
            )}
            data-product-id={productID}
          >
            <SvgIcon
              icon="favorite"
              fillId="white"
              size="xl"
              className={themeData.classes.favoriteIcon}
            />
          </div>
        </div>
      )}

      <div className={themeData.classes.headerWrapper}>
        {fields.eyebrowText?.value ? (
          <Eyebrow classes={themeData.classes.eyebrow} {...props} />
        ) : fields.productItem?.fields.productSeries &&
          fields.productItem?.fields?.productSeries?.fields?.productTypeName?.value ? (
          <Text
            tag="h4"
            className={themeData.classes.eyebrow}
            field={{ value: fields.productItem.fields.productSeries.fields.productTypeName.value }}
          />
        ) : (
          <></>
        )}

        <Link
          href={
            fields.cta1Link.value.href ||
            fields.productItem?.fields?.productDetailPageLink?.value?.href ||
            ''
          }
        >
          <a className={themeData.classes.headlineWrapper}>
            {fields.headlineText?.value ? (
              <Headline useTag="h4" classes={themeData.classes.headline} {...props} />
            ) : fields.productItem?.fields.productName ? (
              <Text
                useTag="h4"
                className={themeData.classes.headline}
                field={{ value: fields.productItem.fields.productName.value }}
              />
            ) : (
              <></>
            )}
          </a>
        </Link>

        {/* ratings */}
        <div className={themeData.classes.ratingsAndPriceWrapper}>
          {bazaarvoiceProductId && (
            <div
              className="review flex-[0_0_auto]"
              data-bv-show="inline_rating"
              data-bv-product-id={bazaarvoiceProductId}
              data-bv-seo="false"
            ></div>
          )}
          {/* Price Level */}
          {priceLevel && priceLevel > 0 && (
            <div className={bazaarvoiceProductId ? themeData.classes.priceLevelWrapper : ''}>
              <PriceLevel
                priceLevel={priceLevel}
                priceClasses={themeData.classes.priceTextClasses}
                priceLevelClasses={themeData.classes.priceLevelClasses}
              />
            </div>
          )}
        </div>
      </div>
      <div className="order-2 my-xxs self-stretch">
        <Link
          href={
            fields.cta1Link.value.href ||
            fields.productItem?.fields?.productDetailPageLink?.value?.href ||
            ''
          }
        >
          <a>
            {fields.primaryImage?.value?.src && fields.primaryImageMobile?.value?.src ? (
              <ImagePrimary {...props} />
            ) : primaryImageVal || primaryImageMobileVal ? (
              <ImagePrimary
                imageLayout={'intrinsic'}
                fields={{
                  primaryImageCaption: {
                    value: '',
                  },
                  primaryImage: { value: primaryImageVal },
                  primaryImageMobile: { value: primaryImageMobileVal },
                  primaryImageMobileFocusArea:
                    fields.productItem?.fields?.primaryImageMobileFocusArea &&
                    fields.productItem?.fields?.primaryImageMobileFocusArea?.targetItem?.value.value
                      ? {
                          id: '',
                          url: '',
                          name: fields.productItem.fields.primaryImageMobileFocusArea.targetItem
                            .value.value,
                          displayName:
                            fields.productItem.fields.primaryImageMobileFocusArea.targetItem.value
                              .value,
                          fields: {
                            Value: {
                              value:
                                fields.productItem.fields.primaryImageMobileFocusArea.targetItem
                                  .value.value,
                            },
                          },
                        }
                      : {
                          id: '',
                          url: '',
                          name: 'Center',
                          displayName: 'Center',
                          fields: {
                            Value: {
                              value: 'center',
                            },
                          },
                          templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                          templateName: 'Enum',
                        },
                }}
              />
            ) : (
              <></>
            )}
          </a>
        </Link>
      </div>
      {fields.body?.value ? (
        <BodyCopy classes={themeData.classes.body} {...props} />
      ) : fields.productItem?.fields.productDescription ? (
        <RichTextWrapper
          field={{ value: fields.productItem.fields.productDescription.value }}
          classes={themeData.classes.body}
        />
      ) : (
        <></>
      )}

      {themeName === 'aw' ? (
        <div className={themeData.classes.awColorsandCTA}>
          <ColorAndActions />
        </div>
      ) : (
        <ColorAndActions />
      )}
    </div>
  );
};

export default withDatasourceCheck()<ProductPreviewCardProps>(ProductPreviewCard);
