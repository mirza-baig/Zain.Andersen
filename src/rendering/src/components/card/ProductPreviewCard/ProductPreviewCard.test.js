// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import ProductPreviewCard from './ProductPreviewCard';
import defaultData from './ProductPreviewCard.mock-data';

it('renders correctly', async () => {
  const component = snapshot(ProductPreviewCard, { componentProps: defaultData });
  await hasDataComponent(component, 'card/productpreview');
});
