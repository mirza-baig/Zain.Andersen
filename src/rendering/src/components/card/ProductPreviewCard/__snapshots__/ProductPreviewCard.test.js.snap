// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="flex flex-col items-start border border-gray h-full relative"
  data-component="card/productpreview"
>
  <div
    class="absolute top-0 right-0"
  >
    <div
      class="favorite-product absolute right-0 top-0 inline h-0 w-0 cursor-pointer border-t-0 border-l-0 border-r-[60px] border-b-[60px] border-solid border-[transparent_#e3e3e3_transparent_transparent] transition-[border-color]  duration-500 ease-[ease]"
      data-product-id="1001"
    >
      <span
        class="absolute -right-[51px] top-[11px]"
      />
    </div>
  </div>
  <div
    class="flex flex-col items-start px-s pb-s pt-m gap-2 order-1 w-full"
  >
    <a
      class="order-2 font-heavy font-sans text-xs"
      href="/"
    >
      <div
        class="items-top font-sans [&_a:hover]:underline text-xs font-heavy hover:underline "
      >
        <h4>
          Unique Opening Architectures
        </h4>
      </div>
    </a>
    <div
      class="flex items-center order-3"
    >
      
      <div
        class=""
      >
        <div
          class="font-sans text-gray text-xs"
        >
          <span
            class="text-black font-heavy"
          >
            $
          </span>
          <span
            class="text-black font-heavy"
          >
            $
          </span>
          <span
            class="text-black font-heavy"
          >
            $
          </span>
          <span
            class=""
          >
            $
          </span>
          <span
            class=""
          >
            $
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="order-2 my-xxs self-stretch"
  >
    <a
      href="/"
    >
      <div
        class="relative hidden md:block  "
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27602%27%20height=%27368%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="door"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="relative w-full md:hidden"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27602%27%20height=%27368%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="door"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
          />
          <noscript />
        </span>
      </div>
    </a>
  </div>
  <div
    class="px-s my-s order-3 text-dark-gray"
  >
    <div
      class="body-copy"
    >
      <div>
        A odio eius esse soluta. Quia ipsam laborum dolorem eum et asperiores architecto voluptate. Exercitationem magnam qui aperiam rerum. A consequuntur id nihil et aut ab ut quos aut.
      </div>
    </div>
  </div>
  <div
    class="flex max-lg:flex-wrap  lg:mb-s lg:items-end justify-between w-full mt-auto max-lg:mb-s order-3"
  >
    <div
      class="px-xs max-lg:mb-s"
    >
      <h4
        class="text-small"
      >
        Exterior Colors
      </h4>
      <a
        href="/"
      >
        <div
          class="mt-xxxs flex"
        >
          <div
            class="h-[30px] w-[30px] rounded-full mr-[14px] [&_*]:h-[30px] [&_*]:w-[30px] [&_img]:rounded-full"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
              />
              <img
                alt="sandtone-exterior-color-swatch"
                data-nimg="responsive"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="h-[30px] w-[30px] rounded-full mr-[14px] [&_*]:h-[30px] [&_*]:w-[30px] [&_img]:rounded-full"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
              />
              <img
                alt="terratone-exterior-80x80 (1)"
                data-nimg="responsive"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="h-[30px] w-[30px] rounded-full mr-[14px] [&_*]:h-[30px] [&_*]:w-[30px] [&_img]:rounded-full"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
              />
              <img
                alt="white-exterior-80x80 (1)"
                data-nimg="responsive"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
              />
              <noscript />
            </span>
          </div>
          <div>
            +
            2
          </div>
        </div>
      </a>
    </div>
    <div
      class="px-xxs md:px-s order-5 max-md:[&_a]:px-xxs [&_a]:mr-0 max-md:mt-auto"
    >
      <div
        class="mb-0 mb-s flex items-start md:flex-row md:items-center md:space-x-4"
      >
        <a
          aria-label="Default state"
          class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed mb-0 mr-m"
          data-gtm-click=""
          data-gtm-dl-event="cta_click"
          href="/"
          role="link"
          tabindex="0"
          target=""
          title=""
        >
          Default state
          <span
            class="ml-xxs"
          />
        </a>
        <a
          aria-label="Default state"
          class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-gray px-m py-[9px] font-sans text-button font-heavy hover:border-black hover:bg-black hover:text-white disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
          data-gtm-click=""
          data-gtm-dl-event="cta_click"
          href="/Demo-HeadlineTwoColumn"
          role="link"
          tabindex="0"
          target=""
          title=""
        >
          Default state
          <span
            class="ml-xxs"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
